go/libraries/doltcore/sqle/tables.go

```diff
@@ -608,12 +608,50 @@ func (t *WritableDoltTable) AutoIncrementSetter(ctx *sql.Context) sql.AutoIncrem
 
 // GetAutoIncrementValue gets the last AUTO_INCREMENT value
 func (t *WritableDoltTable) GetAutoIncrementValue(ctx *sql.Context) (interface{}, error) {
+	sess := DSessFromSess(ctx.Session)
+
 	if !t.autoIncCol.AutoIncrement {
 		return nil, sql.ErrNoAutoIncrementCol
 	}
+
+	autoIncTracker, foundAit := sess.GetDoltDbAutoIncrementTracker(t.db.Name())
+	if !foundAit || autoIncTracker == nil {
+		return t.getAutoIncrementValue(ctx)
+	}
+
+	stored, err := t.getAutoIncrementValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ok, err := autoIncTracker.Request(t.tableName, stored)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the current session cannot requests its current key than ask for the next one and request again.
+	if !ok {
```

Comment:
This isn't right, it only works if exactly one other thread has requested a key. This needs to be a loop.

(or, per above comment, make the tracker smart so that Next() always just works)